pkg/apis/pxc/v1/pxc_types.go

```diff
@@ -525,22 +554,15 @@ func (cr *PerconaXtraDBCluster) CheckNSetDefaults(serverVersion *version.ServerV
 		}
 	}
 
-	if cr.Spec.UpdateStrategy == SmartUpdateStatefulSetStrategyType &&
-		(cr.Spec.ProxySQL == nil || !cr.Spec.ProxySQL.Enabled) &&
-		(cr.Spec.HAProxy == nil || !cr.Spec.HAProxy.Enabled) {
-		return false, fmt.Errorf("ProxySQL or HAProxy should be enabled if SmartUpdate set")
-	}
-
-	if len(cr.Spec.Platform) == 0 {
+	if len(c.Platform) == 0 {
 		if len(serverVersion.Platform) > 0 {
-			cr.Spec.Platform = serverVersion.Platform
+			c.Platform = serverVersion.Platform
 		} else {
-			cr.Spec.Platform = version.PlatformKubernetes
+			c.Platform = version.PlatformKubernetes
 		}
 	}
 
 	cr.setSecurityContext()
-	cr.Spec.Platform = serverVersion.Platform
```

Comment:
why we remove it?

Reply:
Because we have before 
```
if len(c.Platform) == 0 {
		if len(serverVersion.Platform) > 0 {
			c.Platform = serverVersion.Platform
		} else {
			c.Platform = version.PlatformKubernetes
		}
	}
```
Which is actually sets cr.Spec.Platform if it's empty
Is i'm missing something?